common/djangoapps/third_party_auth/tasks.py

```diff
@@ -36,7 +37,7 @@ def fetch_saml_metadata():
         num_failed: Number of providers that could not be updated
         num_total: Total number of providers whose metadata was fetched
     """
-    if not SAMLConfiguration.is_enabled():
+    if not SAMLConfiguration.is_enabled(get_current_site_default()):
```

Comment:
Okay, seems like we would never be able to run this then for sites other than the default site. Does this matter? It seems IdPs associated with site other than the default site would never have their metadata updated. Let me know if I am missing something.